subsys/net/lib/http/http_client.c

```diff
@@ -218,11 +218,16 @@ static int on_header_field(struct http_parser *parser, const char *at,
 						struct http_request,
 						internal.parser);
 	const char *content_len = "Content-Length";
-	uint16_t len;
+	const char *content_range = "Content-Range";
```

Comment:
I think we could change this to 
`const char content_range[]="Content-Range";`
This way we avoid allocating a pointer and can use sizeof instead of strlen
Same change could be done for content-lenght.
Perhaps it could be even static.